Concurrent modification exception in MetricPRocessEventListener

### Describe the bug

Users of metrics are seeing this exception
```
at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1221)
	at org.kie.kogito.monitoring.core.common.process.MetricsProcessEventListener.getRunningProcessInstancesGauge(MetricsProcessEventListener.java:111)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/addons/common/monitoring/core/src/main/java/org/kie/kogito/monitoring/core/common/process/MetricsProcessEventListener.java
+++ b/addons/common/monitoring/core/src/main/java/org/kie/kogito/monitoring/core/common/process/MetricsProcessEventListener.java
@@ -18,8 +18,8 @@
  */
 package org.kie.kogito.monitoring.core.common.process;
 
-import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
@@ -49,7 +49,7 @@ import io.micrometer.core.instrument.Tag;
 public class MetricsProcessEventListener extends DefaultKogitoProcessEventListener {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MetricsProcessEventListener.class);
-    private static Map<String, AtomicInteger> gaugeMap = new HashMap<>();
+    private static Map<String, AtomicInteger> gaugeMap = new ConcurrentHashMap<>();
     private final String identifier;
     private final KogitoGAV gav;
     private final MeterRegistry meterRegistry;

